common/persistence/persistenceTestBase.go

```diff
@@ -106,66 +109,102 @@ func newTestShardContext(shardInfo *ShardInfo, transferSequenceNumber int64, his
 	}
 }
 
-func (s *testShardContext) GetExecutionManager() ExecutionManager {
+// GetExecutionManager test implementation
+func (s *TestShardContext) GetExecutionManager() ExecutionManager {
 	return s.executionMgr
 }
 
-func (s *testShardContext) GetHistoryManager() HistoryManager {
+// GetHistoryManager test implementation
+func (s *TestShardContext) GetHistoryManager() HistoryManager {
 	return s.historyMgr
 }
 
-func (s *testShardContext) GetNextTransferTaskID() (int64, error) {
+// GetNextTransferTaskID test implementation
+func (s *TestShardContext) GetNextTransferTaskID() (int64, error) {
 	return atomic.AddInt64(&s.transferSequenceNumber, 1), nil
 }
 
-func (s *testShardContext) GetTransferMaxReadLevel() int64 {
+// GetTransferMaxReadLevel test implementation
+func (s *TestShardContext) GetTransferMaxReadLevel() int64 {
 	return atomic.LoadInt64(&s.transferSequenceNumber)
 }
 
-func (s *testShardContext) GetTransferAckLevel() int64 {
+// GetTransferAckLevel test implementation
+func (s *TestShardContext) GetTransferAckLevel() int64 {
 	return atomic.LoadInt64(&s.shardInfo.TransferAckLevel)
 }
 
-func (s *testShardContext) GetTimerSequenceNumber() int64 {
-	return atomic.AddInt64(&s.timerSequeceNumber, 1)
-}
-
-func (s *testShardContext) UpdateAckLevel(ackLevel int64) error {
+// UpdateTransferAckLevel test implementation
+func (s *TestShardContext) UpdateTransferAckLevel(ackLevel int64) error {
 	atomic.StoreInt64(&s.shardInfo.TransferAckLevel, ackLevel)
 	return nil
 }
 
-func (s *testShardContext) GetTransferSequenceNumber() int64 {
+// GetTransferSequenceNumber test implementation
+func (s *TestShardContext) GetTransferSequenceNumber() int64 {
 	return atomic.LoadInt64(&s.transferSequenceNumber)
 }
 
-func (s *testShardContext) CreateWorkflowExecution(request *CreateWorkflowExecutionRequest) (
+// GetTimerAckLevel test implementation
+func (s *TestShardContext) GetTimerAckLevel() time.Time {
+	s.Lock()
+	defer s.Unlock()
+	return s.shardInfo.TimerAckLevel
+}
+
+// UpdateTimerAckLevel test implementation
+func (s *TestShardContext) UpdateTimerAckLevel(ackLevel time.Time) error {
+	s.Lock()
+	defer s.Unlock()
+	s.shardInfo.TimerAckLevel = ackLevel
+	return nil
+}
+
+// CreateWorkflowExecution test implementation
+func (s *TestShardContext) CreateWorkflowExecution(request *CreateWorkflowExecutionRequest) (
 	*CreateWorkflowExecutionResponse, error) {
 	return s.executionMgr.CreateWorkflowExecution(request)
 }
 
-func (s *testShardContext) UpdateWorkflowExecution(request *UpdateWorkflowExecutionRequest) error {
+// UpdateWorkflowExecution test implementation
+func (s *TestShardContext) UpdateWorkflowExecution(request *UpdateWorkflowExecutionRequest) error {
```

Comment:
Transfer task IDs are spread across all create/update/activity or decision task creation methods, after moving transfer task IDs to shardContext, the test code is not update. at some time that logic needs to be moved to one place here. i don't want to change transfer task related code as part of this. a later change.